adapters/ix/ix_test.go

```diff
@@ -101,3 +103,99 @@ func TestIxMakeBidsWithCategoryDuration(t *testing.T) {
 		t.Errorf("should not have any errors, errors=%v", errors)
 	}
 }
+
+func TestBuildIxDiag(t *testing.T) {
+	testCases := []struct {
+		description     string
+		request         *openrtb2.BidRequest
+		expectedRequest *openrtb2.BidRequest
+		expectError     bool
+		pbsVersion      string
+	}{
+		{
+			description: "Base Test",
+			request: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}}}`),
+			},
+			expectedRequest: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}},"ixdiag":{"pbsv":"1.880","pbjsv":"7.20"}}`),
+			},
+			expectError: false,
+			pbsVersion:  "1.880-abcdef",
+		},
+		{
+			description: "No Ext",
+			request: &openrtb2.BidRequest{
+				ID: "1",
+			},
+			expectedRequest: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":null,"ixdiag":{"pbsv":"1.880"}}`),
+			},
+			expectError: false,
+			pbsVersion:  "1.880-abcdef",
+		},
+		{
+			description: "PBS Version no Hyphen",
+			request: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}}}`),
+			},
+			expectedRequest: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}},"ixdiag":{"pbsv":"1.880","pbjsv":"7.20"}}`),
+			},
+			expectError: false,
+			pbsVersion:  "1.880",
+		},
+		{
+			description: "PBS Version empty string",
+			request: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}}}`),
+			},
+			expectedRequest: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":{"channel":{"name":"web","version":"7.20"}},"ixdiag":{"pbjsv":"7.20"}}`),
+			},
+			expectError: false,
+			pbsVersion:  "",
+		},
+		{
+			description: "Error Test",
+			request: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: json.RawMessage(`{"prebid":"channel":{"name":"web","version":"7.20"}}}`),
+			},
+			expectedRequest: &openrtb2.BidRequest{
+				ID:  "1",
+				Ext: nil,
+			},
+			expectError: true,
+			pbsVersion:  "1.880-abcdef",
+		},
+	}
+
+	for _, test := range testCases {
+		version.Ver = test.pbsVersion
```

Comment:
Can we reset `version.Ver` back to its original empty `""` value please?
```
181       for _, test := range testCases {
182           version.Ver = test.pbsVersion
183           err := BuildIxDiag(test.request)
184           if test.expectError {
185               assert.NotNil(t, err)
186           } else {
187               assert.Equal(t, test.request, test.expectedRequest)
188               assert.Nil(t, err)
189           }
190       }
    +         version.Ver = ""
191   }
```

Reply:
yep updated